lldb/source/Plugins/ABI/LoongArch/ABISysV_loongarch.cpp

```diff
@@ -0,0 +1,672 @@
+        eRegisterKindGeneric, LLDB_REGNUM_GENERIC_ARG1 + idx);
+    LLDB_LOG(log, "About to write arg{0} (0x{1:x}) into {2}", idx, arg,
+             reg_info->name);
+
+    if (!reg_ctx_sp->WriteRegisterFromUnsigned(reg_info, arg)) {
+      LLDB_LOG(log, "Failed to write arg{0} (0x{1:x}) into {2}", idx, arg,
+               reg_info->name);
+      return false;
+    }
+  }
+
+  if (!UpdateRegister(reg_ctx_sp.get(), eRegisterKindGeneric,
+                      LLDB_REGNUM_GENERIC_PC, func_addr))
+    return false;
+  if (!UpdateRegister(reg_ctx_sp.get(), eRegisterKindGeneric,
+                      LLDB_REGNUM_GENERIC_SP, sp))
+    return false;
+  if (!UpdateRegister(reg_ctx_sp.get(), eRegisterKindGeneric,
+                      LLDB_REGNUM_GENERIC_RA, return_addr))
+    return false;
+
+  LLDB_LOG(log, "ABISysV_riscv::{0}() success", __FUNCTION__);
+  return true;
+}
+
+bool ABISysV_loongarch::GetArgumentValues(Thread &thread,
+                                          ValueList &values) const {
+  // TODO: Implement
+  return false;
+}
+
+Status ABISysV_loongarch::SetReturnValueObject(StackFrameSP &frame_sp,
+                                               ValueObjectSP &new_value_sp) {
+  Status result;
+  if (!new_value_sp) {
+    result = Status::FromErrorString("Empty value object for return value.");
+    return result;
+  }
+
+  CompilerType compiler_type = new_value_sp->GetCompilerType();
+  if (!compiler_type) {
+    result = Status::FromErrorString("Null clang type for return value.");
+    return result;
+  }
+
+  auto &reg_ctx = *frame_sp->GetThread()->GetRegisterContext();
+
+  bool is_signed = false;
+  if (!compiler_type.IsIntegerOrEnumerationType(is_signed) &&
+      !compiler_type.IsPointerType()) {
+    result = Status::FromErrorString(
+        "We don't support returning other types at present");
+    return result;
+  }
+
+  DataExtractor data;
+  size_t num_bytes = new_value_sp->GetData(data, result);
+
+  if (result.Fail()) {
+    result = Status::FromErrorStringWithFormat(
+        "Couldn't convert return value to raw data: %s", result.AsCString());
+    return result;
+  }
+
+  size_t reg_size = m_is_la64 ? 8 : 4;
+  // Currently, we only support sizeof(data) <= 2 * reg_size.
+  // 1. If the (`size` <= reg_size), the `data` will be returned through `ARG1`.
+  // 2. If the (`size` > reg_size && `size` <= 2 * reg_size), the `data` will be
+  // returned through a pair of registers (ARG1 and ARG2), and the lower-ordered
+  // bits in the `ARG1`.
+  if (num_bytes > 2 * reg_size) {
+    result = Status::FromErrorString(
+        "We don't support returning large integer values at present.");
+    return result;
+  }
+
+  offset_t offset = 0;
+  uint64_t raw_value = data.GetMaxU64(&offset, num_bytes);
+  auto reg_info =
+      reg_ctx.GetRegisterInfo(eRegisterKindGeneric, LLDB_REGNUM_GENERIC_ARG1);
+  if (!reg_ctx.WriteRegisterFromUnsigned(reg_info, raw_value)) {
```

Comment:
According to psABI, i32 (no matter signed or unsigned) should be sign-extended in register.